database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala

```diff
@@ -10,23 +10,36 @@ trait JobStoreSlickDatabase extends JobStoreSqlDatabase {
 
   import dataAccess.driver.api._
 
-  override def addJobStoreEntries(entries: Iterable[JobStoreEntry])(implicit ec: ExecutionContext): Future[Unit] = {
-    val action = dataAccess.jobStoreAutoInc ++= entries
+  private def addJobStoreEntry(jobStoreEntry: JobStoreEntry, jobStoreSimpletonEntries: Iterable[JobStoreResultSimpletonEntry])
+                              (implicit ec: ExecutionContext): DBIO[Unit] = {
+    for {
+      insertJobStoreEntry <- dataAccess.jobStoreAutoInc += jobStoreEntry
+      entries = jobStoreSimpletonEntries map { _.copy(jobStoreId = insertJobStoreEntry.self) }
```

Comment:
OK I think this may be a dependent typing thing; `Int` appears to be the type of `RU` in `ReturningInsertActionComposer[U, RU]` after the `returning jobStore.map(_.jobStoreId)` business.  

I have no idea how I went down that `self` path, that appears to be impliciting the `Int` into a `RichInt` and then pulling the raw `Int` back out.  I think I'll just go with your version instead.  😄